NextOfKin/Classes/KinSdkController.swift

```diff
@@ -20,87 +22,52 @@ public class KinSdkController: KinControllerType, KinRespositoryType {
-    private let disposeBag = DisposeBag()
-    private let kinOperationScheduler = SerialDispatchQueueScheduler(internalSerialQueueName: "com.kin.KinSdkController.KinOperationQueue")
-    private var kinClient: KinClient!
+    private let serviceProvider: NOKServiceProvider
 
-    public convenience init() {
-        self.init(keychain: Keychain(service: String(format: "%@.kinSdkController.keychain.key", Bundle.bundleName())),
-                  defaults: UserDefaults.standard)
+    public convenience init(usingTestNet: Bool = true) {
+        let keychain = Keychain(service: String(format: "%@.kinSdkController.keychain.key", Bundle.bundleName()))
+        self.init(keychain: keychain,
+                  usingTestNet: usingTestNet)
     }
 
-    public required init(keychain: Keychain, defaults: UserDefaults) {
+    public required init(keychain: Keychain, usingTestNet: Bool = true) {
         self.keychain = keychain
-        self.defaults = defaults
         
-        // MARK: forced on test network
-        self.defaults.set(Constants.providerUrlOptions[1],
-                          forKey: Constants.configKinProviderUrl)
+        if usingTestNet {
+            self.serviceProvider = NOKServiceProvider(urlString: Constants.testNet.url,
+                                                      networkId: Constants.testNet.network)
+        } else {
+            self.serviceProvider = NOKServiceProvider(urlString: Constants.mainNet.url,
+                                                      networkId: Constants.mainNet.network)
+        }
         
         // load initial existence of an account
         getClient().subscribe(onSuccess: { [weak self] (client) in
             guard let this = self else {
                 return
             }
             this.account.onNext(client.account);
-        })
-            .disposed(by: self.disposeBag)
-    }
-    
-    public var providerUrl: String {
-        guard let providerUrl: String = defaults.string(forKey: Constants.configKinProviderUrl) else {
-            return "nok://undefined"
-        }
-        return providerUrl
+        }).disposed(by: self.disposeBag)
     }
 
     /**
-     * @return If the provided URL is approved to access the Kin SDK
+     * @return The current provider Url used to access the kin network
      */
-    public func isAllowedAccess(with url: URL) -> Bool {
-        guard defaults.bool(forKey: Constants.isKinWalletRestricted) else { // TODO: Should "restricted" not be the default case?
-            // allow all domains to pass the check
-            return true
-        }
-
-        // only allow the current wallet site, served over HTTPS to access restricted plugin methods
-        let isValidHost = url.host == providerUrl
-        let isValidScheme = url.scheme?.lowercased() == "https"
-
-        return isValidHost && isValidScheme
+    public var providerUrl: String {
+        return serviceProvider.url.description
```

Comment:
.description really? shouldn't it be absoluteString? Do we need to expose this as a string as well? might be better to expose as url and then convert to string as needed.

Reply:
also, why a var, shouldn't it be let since we set it once during init?

Reply:
Computed properties have to use var